targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/watchdog.c

```diff
@@ -0,0 +1,111 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "watchdog_api.h"
+
+#include "reset_reason_api.h"
+
+#include "fsl_wdog.h"
+
+// Platform specific watchdog definitions
+#define LPO_CLOCK_FREQUENCY 1000
+#define MAX_PRESCALER       8
+#define MAX_TIMEOUT         0xFFFFFFFFULL
+
+// Number of decrements in the timeout register per millisecond
+#define TICKS_PER_MS (LPO_CLOCK_FREQUENCY / 1000)
+// Maximum timeout that can be specified in milliseconds
+const uint64_t max_timeout_ms = ((MAX_TIMEOUT / TICKS_PER_MS) * MAX_PRESCALER);
+
+// Maximum supported watchdog timeout for given prescaler value
+#define CALCULATE_MAX_TIMEOUT_MS(scale) \
+  ((MAX_TIMEOUT / TICKS_PER_MS) * scale)
+
+
+static uint32_t calculate_prescaler_value(const uint32_t timeout_ms)
+{
+  if (timeout_ms > max_timeout_ms) {
+    return 0;
+  }
+
+  for (uint32_t scale = 1; scale <= MAX_PRESCALER; ++scale) {
+    if (timeout_ms < CALCULATE_MAX_TIMEOUT_MS(scale)) {
```

Comment:
As long as we consider `timeout_ms == max_timeout_ms` to be valid, this condition should be `timeout_ms <= CALCULATE_MAX_TIMEOUT_MS(scale)`.